### PR TITLE
1815 Add train mode and eval mode in Evaluators

### LINKED ISSUE #1815: allows for flexible model "mode" in the evaluators
**Is your feature request related to a problem? Please describe.**
provide an option for all the evaluators to be able to switch between "train_mode" and "eval_mode" of the model.
(this is useful for GradCAM inference integration)

**Additional context**
https://github.com/Project-MONAI/MONAI/blob/70388fa298384f698dfb3ba790e3bedb1ffcb07b/monai/engines/evaluator.py#L200

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #1815 .

### Description
This PR added support to set `train` or `eval` mode in all the evaluators.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/engines/evaluator.py [hunk 1]
```diff
@@ -85,6 +88,12 @@ def __init__(
             handlers=val_handlers,
             amp=amp,
         )
+        if mode == "eval":
```
Comment (ericspod) [inline monai/engines/evaluator.py:91]: "eval" and "train" could be put into an Enum called "TrainMode" or something like that which we might add to later.

**Review by ericspod — APPROVED** [review]

**Comment by ericspod** [comment]
I mentioned something small you might want to change but otherwise good to go.

File: monai/engines/evaluator.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/engines/evaluator.py]: sure, updated in the latest commit.
Thanks.

**Pull request merged**